src/html2ans/parsers/text.py

```diff
@@ -277,30 +314,45 @@ class ListParser(BaseElementParser):
     .. code-block:: python
 
         {
-            "type": "list",
-            "list_type": "unordered",
-            "items": [
+            'type': 'list',
+            'list_type': 'unordered',
+            'items': [
                 {
-                    "type": "text",
-                    "content": "Post Reports is the daily podcast from The Washington Post."
+                    'type': 'text',
+                    'content': 'Post Reports is the daily'
                 },
                 {
-                    "type": "text",
-                    "content": "Unparalleled reporting."
+                    'type': 'text',
+                    'content': '<a href="/podcast/">podcast</a>'
                 },
                 {
-                    "type": "text",
-                    "content": "Expert insight."
+                    'type': 'text',
+                    'content': 'from The Washington Post.'
                 },
                 {
-                    "type": "text",
-                    "content": "Clear analysis."
+                    'type': 'list',
+                    'list_type': 'unordered',
+                    'items': [
+                        {
+                            'type': 'text',
+                            'content': 'Unparalleled reporting.'
+                        },
+                        {
+                            'type': 'text',
+                            'content': 'Expert insight.'
+                        }
+                    ]
+                },
+                {
+                    'type': 'text',
+                    'content': 'Clear analysis.'
                 }
             ]
         }
 
     """
     applicable_elements = ['ul', 'ol']
+    text_parsers = [InlineLinkParser, ParagraphParser]
```

Comment:
My previous comment implies that you only need the `ParagraphParser`. I think adding the capability of multiple parsers here leads to reinventing some of the logic of the `BaseHtmlAnsParser`; I think we should just have one sub-parser here.